### PR TITLE
fix `group by ... having ...` aliasing once more

### PR DESCRIPTION
**Pull request by jycor**
This PR simplifies the `GROUP BY` and `HAVING` aliasing rules to much more closely match MySQL's weirdness.

TODO: Go through various group by aliasing issues and link fixed ones here

correctness: https://github.com/dolthub/dolt/pull/7455

### CODE REVIEW

File: sql/planbuilder/aggregates.go [hunk 1]
```diff
@@ -812,67 +813,55 @@ func (b *Builder) buildHaving(fromScope, projScope, outScope *scope, having *ast
...
-		// if a projection overrides a column used in an aggregation, don't use projection
-		found := false
-		if fromScope.groupBy != nil && fromScope.groupBy.hasAggs() {
-			for _, cc := range fromScope.groupBy.aggregations() {
-				transform.InspectExpr(cc.scalar, func(e sql.Expression) bool {
-					switch e := e.(type) {
-					case *expression.GetField:
-						if strings.EqualFold(c.col, e.Name()) {
-							found = true
+	havingScope := b.newScope()
```
Comment (max-hoffman) [inline sql/planbuilder/aggregates.go:817]: could you add a docstring summarizing top-level rules? The before/after looks mostly the same just in a different order

File: enginetest/queries/script_queries.go [hunk 2]
```diff
@@ -5277,6 +5277,134 @@ CREATE TABLE tab3 (
...
+		Name: "group by having with conflicting aliases test",
+		SetUpScript: []string{
+			"CREATE TABLE tab2(col0 INTEGER, col1 INTEGER, col2 INTEGER);",
+			"INSERT INTO tab2 VALUES(15,61,87);",
+			"INSERT INTO tab2 VALUES(91,59,79);",
+			"INSERT INTO tab2 VALUES(92,41,58);",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: `SELECT - col2 AS col0 FROM tab2 GROUP BY col0, col2 HAVING NOT + + col2 <= - col0;`,
```
Comment (max-hoffman) [inline enginetest/queries/script_queries.go:5290]: would any subset of these be useful planbuilder tests that verify refs get the right definition id?

Comment (jycor) [reply enginetest/queries/script_queries.go:5290]: added some of these to planbuilder

**Review by max-hoffman — APPROVED** [review]
lgtm, the having scope logic is a lot clearer now. Easier to extend in the future if there are extra edge cases

**Pull request merged**